### PR TITLE
allow image requests to have background priority

### PR DESCRIPTION
**Pull request by nwadams**
background priority support
@lxdvs

### CODE REVIEW

**Comment by lxdvs** [comment]
:+1: looks reasonable

File: src/com/android/volley/Request.java [hunk 1]
```diff
@@ -539,6 +539,7 @@ public final boolean shouldCache() {
      * lower priorities, in FIFO order.
      */
     public enum Priority {
+        BACKGROUND,
```
Comment (geo-opensource) [inline src/com/android/volley/Request.java:542]: do we really need a lower-than-low priority?

File: src/com/android/volley/toolbox/ImageLoader.java [hunk 2]
```diff
@@ -165,6 +165,25 @@ public boolean isCached(String requestUrl, int maxWidth, int maxHeight) {
...
     /**
+     * get Image for background caching
+     * @param requestUrl
+     * @return
+     */
+    public ImageContainer getImageBackground(String requestUrl) {
+        return get(requestUrl, new ImageListener() {
+            @Override
+            public void onResponse(ImageContainer response, boolean isImmediate) {
+
```
Comment (geo-opensource) [inline src/com/android/volley/toolbox/ImageLoader.java:176]: comment why we have empty listener?
or maybe check for a non-null listener inside get()

File: src/com/android/volley/toolbox/ImageLoader.java [hunk 3]
```diff
@@ -196,7 +220,7 @@ public ImageContainer get(String requestUrl, ImageListener imageListener,
      *     the currently available image (default if remote is not loaded).
      */
     public ImageContainer get(String requestUrl, ImageListener imageListener,
```
Comment (geo-opensource) [inline src/com/android/volley/toolbox/ImageLoader.java:223]: can you add documentation on backgroundFetch (just say to pass false for normal requests, true if it's to pre-fetch something in the background). and on doublerespond if you'd like (but that is more clear)

File: src/com/android/volley/Request.java (hunk 1 above)
Comment (nwadams) [inline src/com/android/volley/Request.java:542]: we do. we want offline sync to be lower than images.

File: src/com/android/volley/toolbox/ImageLoader.java (hunk 2 above)
Comment (nwadams) [inline src/com/android/volley/toolbox/ImageLoader.java:176]: image loader requires non empty listener. Maybe we can fix that.

File: src/com/android/volley/toolbox/ImageLoader.java (hunk 3 above)
Comment (nwadams) [inline src/com/android/volley/toolbox/ImageLoader.java:223]: sure.

**Comment by elihart** [comment]
+1

**Pull request merged**